gcloud/datastore/query.py

```diff
@@ -463,6 +463,11 @@ def __iter__(self):
                 yield entity
             if not self._more_results:
                 break
+            num_results = len(self._page)
+            if self._limit is not None:
+                self._limit -= num_results
+            if self._offset is not None:
+                self._offset += num_results
```

Comment:
Got it, thanks. To confirm, the only way "another query" continues the previous one is with the `end_cursor` returned, yes? Concretely, without passing along a cursor:

```python
>>> q = client.query(kind='Foo')
>>> q_it1 = q.fetch(offset=0, limit=3)
>>> list(q_it1)
[<Entity[{'kind': u'Foo', 'id': 4892024121589760L}] {u'bar': 3L}>,
 <Entity[{'kind': u'Foo', 'id': 5173499098300416L}] {u'bar': 1L}>,
 <Entity[{'kind': u'Foo', 'id': 5736449051721728L}] {u'bar': 0L}>]
>>> q_it2 = q.fetch(offset=1, limit=2)
>>> list(q_it2)
[<Entity[{'kind': u'Foo', 'id': 5173499098300416L}] {u'bar': 1L}>,
 <Entity[{'kind': u'Foo', 'id': 5736449051721728L}] {u'bar': 0L}>]
```